SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -744,4 +761,19 @@ private int numInstancesExpected(SingularityRequest request, SingularityPendingR
     return Optional.of(nextRunAt);
   }
 
+  private int checkStartupThreshold(SingularityPendingRequest pendingRequest, int totalTasks) {
+    if (pendingRequest.getPendingType() == PendingType.STARTUP) {
+      int numNewStartupTasks = taskManager.getNumActiveTasksForRequest(pendingRequest.getRequestId());
```

Comment:
this isn't quite what you want. `getActiveTasksForRequest` will get you _all_ active tasks for that request, so we might be including ones that were already running here. We already have the `numScheduledTasks` value a few lines above where this is called in `drainPendingQueue`. That value is just the number of tasks that were started due to this particular request and is what you want to be counting